Objects/moduleobject.c

```diff
@@ -88,21 +88,31 @@ new_module_notrack(PyTypeObject *mt)
     m->md_weaklist = NULL;
     m->md_name = NULL;
     m->md_dict = PyDict_New();
-    if (m->md_dict != NULL) {
-        return m;
+    if (m->md_dict == NULL) {
+        Py_DECREF(m);
+        return NULL;
     }
-    Py_DECREF(m);
-    return NULL;
+    return m;
+}
+
+static void
+track_module(PyModuleObject *m)
+{
+    _PyObject_SetDeferredRefcount(m->md_dict);
+    PyObject_GC_Track(m->md_dict);
```

Comment:
There's part of the code that treat `md_dict` as possibly not being initialized at this stage e.g. `module___init___impl` but I don't think that's possible here since `new_module_notrack` ensures a dictionary. So this should be alright.

Reply:
I think that's old code that no longer makes sense. https://github.com/python/cpython/pull/26847 in (3.11) made sure that module's always have a valid `md_dict`.

I'll clean it up.